wandb/sdk/launch/agent/agent.py

```diff
@@ -589,6 +589,21 @@ def _check_run_finished(
         try:
             run = job_tracker.run
             status = run.get_status().state
+
+            # Ensure run was not stopped intentionally
+            if status == "preempted":
```

Comment:
when `status == "preempted"` would we expect the `RunState` to be "preempted" or just CommError? Just curious.

Reply:
It depends - if the run was stopped intentionally, eg from the UI, it'll be "stopped". If the run was stopped by something unexpected like a pod going down, the RunState won't have been updated yet, so it'll still be "running".

I realized that after I moved everything into this `if status == "preempted"` block, we shouldn't expect CommErrors anymore, so I'm removing that inner try-block. Before, it was trying to get run status from runs that were still being created.